### PR TITLE
Settings and refactor

### PR DESCRIPTION
**Pull request by keptsecret**
Creates a `lei-config.ini` in the working directory.  
Defaults to

```
# Lei3D config file

# Screen size
width:2560
height:1600
fullscreen:0

fov:55
mousespeed:0.1

# Graphics settings
# shadow quality: 0 - medium, 1 - high
shadowqual:0
# enable/disable screen space reflections
ssr:0
```

Width and height screen sizes are screen resolutions.  
Reads from file once created.

Might wait for more stuff to begin refactor?

### CODE REVIEW

**Comment by keptsecret** [comment]
Added a preprocessor directive `EDITOR` that blocks out any code used only in editor mode  
Editor now compiles to a separate library `leieditor_lib`

Removed initialization and usage of Imgui in game mode  
Also disabled logging in game mode

**Comment by kevinsadi** [comment]
This works great. How is the EDITOR flag being defined when we make a Lei_Editor build?

**Review by kevinsadi — APPROVED** [review]
Aside from my question and the merge conflict, this looks great to me!

File: src/engine/CMakeLists.txt [hunk 1]
```diff
@@ -9,14 +9,41 @@ file(GLOB_RECURSE ENGINE_HEADER
...
+        ${IMGUI_INCLUDE}
+        ${SPDLOG_INCLUDE}
+        ${BULLET_INCLUDE}
+        ${MINIAUDIO_INCLUDE}
+        ${STB_INCLUDE}
+        ${ASSIMP_INCLUDE}
+        ${GLM_INCLUDE}
+        )
+
+target_compile_definitions(leieditor_lib PRIVATE -DEDITOR)
```
Comment (keptsecret) [inline src/engine/CMakeLists.txt:35]: > How is the EDITOR flag being defined when we make a Lei_Editor build?

@kevinsadi Lei_Editor now uses a separate library, compiled with the `EDITOR` flag

Comment (kevinsadi) [reply src/engine/CMakeLists.txt:35]: ah makes sense

**Pull request merged**